src/components/Header.js

```diff
@@ -7,20 +7,25 @@ import {
   Collapse,
   Nav,
   NavLink,
+  Button,
 } from 'reactstrap'
 import { connect } from 'react-redux'
 
 import FontAwesomeIcon from '@fortawesome/react-fontawesome'
 import faUser from '@fortawesome/fontawesome-free-solid/faUser'
 
 import { Link } from '../routes'
+import { baseUrl, isDev } from '../util'
 
 const styles = {
   navbar: {
     zIndex: '10',
   },
 }
 
+const origin = (typeof this !== 'undefined' && this.location.origin) || ''
+const logoutLink = `${origin}${baseUrl}/Shibboleth.sso/Logout`
```

Comment:
So the logout link that @wadefagen sent me (that also worked to log me out of the deployed queue) is `https://edu.cs.illinois.edu/Shibboleth.sso/Logout` I thought that `edu.cs.illinois.edu` was covered by the origin and baseUrl? How would this differ for the host?

Reply:
Shib auth is done on a per-host basis, and our staging environment is `edu-staging.cs.illinois.edu`. It would be very confusing to click "Log out" in the staging website but be logged out of the actual queue! This also applies to cases where we're hypothetically deployed at UBC, for instance.

Reply:
Also sorry, it's late, that didn't fully answer your question. You've got the host thing right by using `location.origin`, but the logout link doesn't need to include the base url. If it did, that would make the logout link `https://edu.cs.illinois.edu/queue/Shibboleth.sso/Logout`, since `baseUrl` is `/queue` in production.